assembly/JSON.ts

```diff
@@ -183,24 +183,33 @@ export abstract class Value {
 }
 
 export class Str extends Value {
+
   constructor(public _str: string) {
     super();
   }
 
+  private escapeChar(char: string): string {
+    const charCode = char.charCodeAt(0);
+    switch (charCode) {
+      case 0x22: return '\\"';
+      case 0x5C: return "\\\\";
+      case 0x08: return "\\b";
+      case 0x0A: return "\\n";
+      case 0x0D: return "\\r";
+      case 0x09: return "\\t";
+      case 0x0C: return "\\f";
+      case 0x0B: return "\\u000b";
+      default: return char;
+    }
+  }
+
   stringify(): string {
-    let escaped: i32[] = [];
+    let escaped: string[] = new Array(this._str.length);
```

Comment:
```suggestion
    let escaped: string[] = new Array(this._str.length * 2); // worst case
```

Reply:
Perhaps we free it before returning the string.

Reply:
I just realized that my point was wrong... since it is an array of strings, then the escaped string "\n" will be replaced by the string "\\n". Therefore, while the number of chars is different, the total number of *strings* is the same, and using `new Array(this._str.length);` is valid.